agent-control/src/opamp/remote_config/validators/signature/validator.rs

```diff
@@ -1,199 +1,117 @@
-    #[serde(default)]
-    pub certificate_pem_file_path: PathBuf,
     #[serde(default = "default_signature_validator_config_enabled")]
     pub enabled: bool,
+    #[serde(default)]
+    pub public_key_server_url: Option<Url>,
 }
-
 impl Default for SignatureValidatorConfig {
     fn default() -> Self {
         Self {
-            enabled: default_signature_validator_config_enabled(),
-            certificate_server_url: default_certificate_server_url(),
+            enabled: DEFAULT_SIGNATURE_VALIDATOR_ENABLED,
             public_key_server_url: None,
-            certificate_pem_file_path: PathBuf::new(),
         }
     }
 }
 
-fn default_certificate_server_url() -> Url {
-    Url::parse(DEFAULT_CERTIFICATE_SERVER_URL).unwrap_or_else(|err| {
-        panic!("Invalid DEFAULT_CERTIFICATE_SERVER_URL: '{DEFAULT_CERTIFICATE_SERVER_URL}': {err}")
-    })
-}
-
 fn default_signature_validator_config_enabled() -> bool {
     DEFAULT_SIGNATURE_VALIDATOR_ENABLED
 }
 
-// NOTE: if we updated the components using the validator to use a composite-like implementation to handle validation,
-// the no-op validator wouldn't be necessary.
-/// The SignatureValidator enum wraps [CertificateSignatureValidator] and adds support for No-op validator.
-#[allow(clippy::large_enum_variant)]
-pub enum SignatureValidator {
-    Composite(CompositeSignatureValidator),
-    Noop,
+pub struct SignatureValidator {
+    public_key_store: Option<VerifierStore<PublicKey, PublicKeyFetcher>>,
 }
 
-impl RemoteConfigValidator for SignatureValidator {
-    type Err = SignatureValidatorError;
-
-    fn validate(
-        &self,
-        agent_identity: &AgentIdentity,
-        opamp_remote_config: &OpampRemoteConfig,
-    ) -> Result<(), Self::Err> {
-        match self {
-            SignatureValidator::Composite(v) => v.validate(agent_identity, opamp_remote_config),
-            SignatureValidator::Noop => Ok(()),
+impl SignatureValidator {
+    pub fn new(
+        config: SignatureValidatorConfig,
+        proxy_config: ProxyConfig,
+    ) -> Result<Self, SignatureValidatorError> {
+        if !config.enabled {
+            warn!("Remote config signature validation is disabled");
+            return Ok(Self::new_noop());
         }
-    }
-}
 
-/// Temporal signature validator that uses both certificate and public key validation in order
-/// to support backward compatibility with existing signatures.
-/// Once the migration to the new signature platform is complete, the certificate validation
-/// can be removed.
-pub struct CompositeSignatureValidator {
-    certificate_store: VerifierStore<Certificate, CertificateFetcher>,
-    public_key_store: Option<VerifierStore<PublicKey, PublicKeyFetcher>>,
-}
+        let Some(public_key_server_url) = config.public_key_server_url else {
+            return Err(SignatureValidatorError::BuildingValidator(
+                "missing public_key_server_url configuration".to_string(),
+            ));
+        };
```

Comment:
We fail if enabled and url is not set, nice!

Reply:
yes , it shouldn't happen on any existing flow becase recipe and helm chart sets the corresponding url but just in case